provider/resource_neap.go

```diff
@@ -0,0 +1,1467 @@
+		"run_time_based_actions_once": obj.RunTimeBasedActionsOnce.ValueBool(),
+		"service_topology_enabled":    obj.ServiceTopologyEnabled.ValueBool(),
+		"entity_factor_enabled":       obj.EntityFactorEnabled.ValueBool(),
+
+		"ace_enabled": 0, //TODO: support smart mode
+	}, nil
+}
+
+func (w *neapBuildWorkflow) episodeInfo(ctx context.Context, obj neapModel) (map[string]interface{}, diag.Diagnostics) {
+	var diags diag.Diagnostics
+
+	episodeSeverity := obj.GroupSeverity.ValueString()
+	episodeStatus := obj.GroupStatus.ValueString()
+
+	var ok bool
+
+	if !neapEpisodeSeverityTemplateValues.Contains(episodeSeverity) {
+		if episodeSeverity, ok = tfToItsiEpisodeSeverityTransform()[episodeSeverity]; !ok {
+			diags.AddError("NEAP: Invalid Episode Severity", fmt.Sprintf("unsupported severity: %s", episodeSeverity))
+		}
+	}
+	if !neapEpisodeStatusTemplateValues.Contains(episodeStatus) {
+		if episodeStatus, ok = tfToItsiEpisodeStatusTransform()[episodeStatus]; !ok {
+			diags.AddError("NEAP: Invalid Episode Status", fmt.Sprintf("unsupported episode status: %s", episodeStatus))
+		}
+	}
+
+	return map[string]interface{}{
+		"group_title":             obj.GroupTitle.ValueString(),
+		"group_description":       obj.GroupDescription.ValueString(),
+		"group_severity":          episodeSeverity,
+		"group_assignee":          obj.GroupAssignee.ValueString(),
+		"group_status":            episodeStatus,
+		"group_instruction":       obj.GroupInstruction.ValueString(),
+		"group_custom_instuction": obj.GroupCustomInstruction.ValueString(),
+		"group_dashboard":         obj.GroupDashboard.ValueString(),
+		"group_dashboard_context": obj.GroupDashboardContext.ValueString(),
+	}, diags
+}
+
+func (w *neapBuildWorkflow) criteria(ctx context.Context, obj neapModel) (map[string]interface{}, diag.Diagnostics) {
+	diags := diag.Diagnostics{}
+
+	var splitByFields []string
+	diags = append(diags, obj.SplitByField.ElementsAs(ctx, &splitByFields, false)...)
+
+	filterCriteria, d := obj.FilterCriteria.apiModel(neapCriteriaTypeFilter)
+	diags.Append(d...)
+
+	breakingCriteria, d := obj.BreakingCriteria.apiModel(neapCriteriaTypeBreaking)
+	diags.Append(d...)
+
+	if diags.HasError() {
+		return nil, diags
+	}
+
+	return map[string]interface{}{
+		"split_by_field":    strings.Join(splitByFields, ","),
+		"filter_criteria":   filterCriteria,
+		"breaking_criteria": breakingCriteria,
+	}, diags
+}
+
+func (w *neapBuildWorkflow) rules(ctx context.Context, obj neapModel) (map[string]interface{}, diag.Diagnostics) {
+	var d, diags diag.Diagnostics
+	rules := make([]map[string]any, len(obj.Rules))
+
+	for i, rule := range obj.Rules {
+		rules[i], d = rule.apiModel()
+		diags.Append(d...)
+	}
+
+	return map[string]interface{}{
+		"rules": rules,
+	}, diags
+}
+
+// (5) [ Neap Parse Workflow ]__________________________________________________
+
+type neapParseWorkflow struct{}
+
+var _ apiparseWorkflow[neapModel] = &neapParseWorkflow{}
+
+//lint:ignore U1000 used by apiparser
```

Comment:
I didn't understand that comment :) I read about linter a bit, didn't use this, will set it up in vscode as well.

Reply:
This comment tells staticcheck (go linter) to ignore the warning that it gives about this function not being used.
<img width="954" alt="Screenshot 2024-04-11 at 8 19 37 AM" src="https://github.com/TiVo/terraform-provider-splunk-itsi/assets/178155/a7b85cc4-57a0-49ae-8bce-b04772feec2e">

Reply:
I think the warning was due to the fact that the function is implementing an interface, as isn't called directly as *neapBuildWorkflow's method, so staticcheck couldn't figure out it's actually being used. So i added the comment for now to ignore that.

Reply:
But yeah, you should use the linter, it actually comes with the standard official VSCode's "Go" extension. I think you just need to run "Go Install/Update Tools" in the VS Code's Command Palette 
<img width="592" alt="Screenshot 2024-04-11 at 8 49 09 AM" src="https://github.com/TiVo/terraform-provider-splunk-itsi/assets/178155/655fef6a-79f4-41d8-8486-9a1c34fbf74d">